crates/ibc/src/vp/mod.rs

```diff
@@ -400,13 +400,15 @@ where
             IbcMessage::NftTransfer(msg_nft_transfer) => {
                 Some(&msg_nft_transfer.message.chan_id_on_a)
             }
-            IbcMessage::Envelope(boxed) => {
-                if let MsgEnvelope::Packet(PacketMsg::Recv(msg)) = &**boxed {
+            IbcMessage::Envelope(boxed) => match &**boxed {
+                MsgEnvelope::Packet(PacketMsg::Recv(msg)) => {
                     Some(&msg.packet.chan_id_on_b)
-                } else {
-                    None
                 }
-            }
+                MsgEnvelope::Packet(PacketMsg::Timeout(msg)) => {
+                    Some(&msg.packet.chan_id_on_a)
+                }
+                _ => None,
+            },
```

Comment:
would there also be an issue with ack errors? on [`PacketMsg::Ack`](https://docs.rs/ibc/latest/ibc/core/channel/types/msgs/enum.PacketMsg.html#variant.Ack)